File: hw/calle/u3/hw29Singleton/PY/utils/rental_manager.py
import os

class RentalManager:
    _instance = None
    FILE_PATH = "rent.txt"

    def __init__(self):
        if RentalManager._instance is not None:
            raise Exception("This class is a singleton!")
        else:
            self.monthly_rent = 20.0
            self._load_rent_from_file()

    @staticmethod
    def get_instance():
        if RentalManager._instance is None:
            RentalManager._instance = RentalManager()
        return RentalManager._instance

    def _load_rent_from_file(self):
        if os.path.exists(self.FILE_PATH):
            try:
                with open(self.FILE_PATH, "r") as f:
                    self.monthly_rent = float(f.read().strip())
            except:
                self.monthly_rent = 20.0

    def update_monthly_rent(self, new_rent):
        self.monthly_rent = new_rent
        with open(self.FILE_PATH, "w") as f:
            f.write(str(new_rent))

    def get_monthly_rent(self):
        return self.monthly_rent

File: hw/calle/u3/hw29Singleton/JS/src/utils/RentalManager.js
//npm init -y
//npm install express
export class RentalManager {
    static instance = null;

    constructor() {
        if (RentalManager.instance) return RentalManager.instance;
        this.monthlyRent = 40.0; 
        RentalManager.instance = this;
    }

    static getInstance() {
        if (!this.instance) this.instance = new RentalManager();
        return this.instance;
    }

    getMonthlyRent() {
        return this.monthlyRent;
    }

    updateMonthlyRent(newValue) {
        if (!isNaN(newValue) && newValue > 0) {
            this.monthlyRent = newValue;
            return true;
        }
        return false;
    }
}